src/OpenTelemetry/Metrics/MetricStreamConfiguration.cs

```diff
@@ -133,6 +133,10 @@ public class MetricStreamConfiguration
         }
     }
 
+    // TODO: Expose this to be complaint with the spec:
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#stream-configuration
+    internal Func<ExemplarReservoir?>? ExemplarReservoirFactory { get; set; }
```

Comment:
any advantage of using a fn/factory as opposed to just `ExemplarReservoir`

Reply:
I think it has to be a factory. Because a view applies to a `Metric` but there are many `MetricPoint`s. If it was a single `ExemplarReservoir` on the view, you would have a single `ExemplarReservoir` shared by all `MetricPoint`s. What the factory lets us do is create a new `ExemplarReservoir` for each `MetricPoint`.

Reply:
makes sense!